packages/labs/gen-wrapper-vue/src/lib/wrapper-module-template.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  LitElementDeclaration,
+  ReactiveProperty as ModelProperty,
+  Event as ModelEvent,
+  PackageJson,
+} from '@lit-labs/analyzer/lib/model.js';
+import {
+  javascript,
+  kabobToOnEvent,
+  toInitialCap,
+} from '@lit-labs/gen-utils/lib/str-utils.js';
+
+export const wrapperModuleTemplate = (
+  packageJson: PackageJson,
+  moduleJsPath: string,
+  elements: LitElementDeclaration[]
+) => {
+  return javascript`
+  import { h, defineComponent, openBlock, createBlock } from "vue";
+  import { wrapSlots, Slots, eventProp } from "@lit-labs/vue-utils/wrapper-utils.js";
+  import '${packageJson.name}/${moduleJsPath}';
+
+  ${elements.map((element) => wrapperTemplate(element))}
+`;
+};
+
+// TODO(sorvell): need to extract Javascript type from typescript for Vue.
+// can workaround this by using an SFC and the macro `defineProperties<>()`
+const tsTypeToVuePropType = (type: string) => {
```

Comment:
This is using Babel's AST, but we could adapt it to tsc's AST: https://github.com/vuejs/core/blob/583b6259870211c32efee0bb4a60b342799d80f7/packages/compiler-sfc/src/compileScript.ts#L1763